nav2_smac_planner/src/a_star.cpp

```diff
@@ -242,14 +294,24 @@ bool AStarAlgorithm<NodeT>::areInputsValid()
   }
 
   // Check if points were filled in
-  if (!_start || !_goal) {
+  if (!_start || _goal_manager.goalsIsEmpty()) {
     throw std::runtime_error("Failed to compute path, no valid start or goal given.");
   }
 
+  const bool allow_infeasible = getToleranceHeuristic() > 0.001;
+
+  // Predicate: should we try this goal?
+  auto shouldTryGoal = [&](const NodePtr & node_ptr) {
+      return node_ptr->isNodeValid(_traverse_unknown, _collision_checker) || allow_infeasible;
+    };
+
+  // Filter and store all tryable goals
+  bool no_tryable_goals = true;
+  _goal_manager.filterAndStoreTryableGoals(shouldTryGoal, no_tryable_goals);
```

Comment:
Why not have a `removeInvalidGoals` API that has the lamba's logic in it? We just pass in the tolerance, collision checker, and traverse unknown members.

Then we can just use the `goals` getter to check if the `size > 0` if for there are any goals left for the `GoalOccupied` check.

Reply:
Making changes, makes sense.

Reply:
Done